### PR TITLE
Adds support for Rails 5 migrations

### PR DESCRIPTION
**Pull request by laertispappas**
Addresses issue #142. Didn't know how to test this any suggestions?

### CODE REVIEW

File: lib/generators/acts_as_votable/migration/migration_generator.rb [hunk 1]
```diff
@@ -24,7 +24,19 @@ def self.next_migration_number(path)
...
       if self.class.orm_has_migration?
-        migration_template 'migration.rb', 'db/migrate/acts_as_votable_migration.rb'
+        migration_template 'migration.erb', 'db/migrate/acts_as_votable_migration.rb', migration_version: migration_version
+      end
+    end
+
+
+    private
+    def rails5?
+      Rails.version.start_with? '5'
```
Comment (marcbest) [inline lib/generators/acts_as_votable/migration/migration_generator.rb]: Could this be made more generic for future versions?

```
Rails.version.to_i >= 5
```

Comment (laertispappas) [inline lib/generators/acts_as_votable/migration/migration_generator.rb]: Sure I will, I will fix the specs as well.

**Comment by epergo** [comment]
Hello! I'm not sure if this is the change needed. If I am not mistaken we have to specify the Rails version where the migration file was _created_ / _generated_ 

This migration file was created in Rails 3, but it works perfectly fine in Rails 3 and 4.

So this file should specify a version less than 5, 3 or 4, but not 5.

What you think?

**Comment by laertispappas** [comment]
@epergo I don't think I did understand you. Above I am passing rails version to `Rails::Generators::Migration` config param in order to generate the correct migration version only if Rails version start with 5. In Rail 5.0 for example an `ActsAsVotableMigration` migration class will be created inheriting from `ActiveRecord::Migration[5.0]`. On a Rails 4 application the corresponding migration class will inherit from `ActiveRecord::Migration` directly.

**Comment by epergo** [comment]
Hi @laertispappas,

The version specified in the header of the file indicates the compatibility mode which will be used to run the migration.

If you use version [5.0] the SQL generated will be different that the SQL generated for version [4.2], examples:

Rails 5 with migration tagged as [5.0]

```ruby
CREATE TABLE "votes" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "votable_type" varchar, "votable_id" integer, "voter_type" varchar, "voter_id" integer, "vote_flag" boolean, "vote_scope" varchar, "vote_weight" integer, "created_at" datetime NOT NULL, "updated_at" datetime NOT NULL)

CREATE  INDEX "index_votes_on_votable_type_and_votable_id" ON "votes" ("votable_type", "votable_id")

CREATE  INDEX "index_votes_on_voter_id_and_voter_type_and_vote_scope" ON "votes" ("voter_id", "voter_type", "vote_scope")

CREATE  INDEX "index_votes_on_votable_id_and_votable_type_and_vote_scope" ON "votes" ("votable_id", "votable_type", "vote_scope")
```

Rails 5 with migration tagged as [4.2]

```ruby
CREATE TABLE "votes" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "votable_type" varchar, "votable_id" integer, "voter_type" varchar, "voter_id" integer, "vote_flag" boolean, "vote_scope" varchar, "vote_weight" integer, "created_at" datetime, "updated_at" datetime)

CREATE  INDEX "index_votes_on_voter_id_and_voter_type_and_vote_scope" ON "votes" ("voter_id", "voter_type", "vote_scope")

CREATE  INDEX "index_votes_on_votable_id_and_votable_type_and_vote_scope" ON "votes" ("votable_id", "votable_type", "vote_scope")
```

Rails 4.2.9 (latest stable)

```ruby
CREATE TABLE "votes" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "votable_id" integer, "votable_type" varchar, "voter_id" integer, "voter_type" varchar, "vote_flag" boolean, "vote_scope" varchar, "vote_weight" integer, "created_at" datetime, "updated_at" datetime)

CREATE  INDEX "index_votes_on_voter_id_and_voter_type_and_vote_scope" ON "votes" ("voter_id", "voter_type", "vote_scope")

CREATE  INDEX "index_votes_on_votable_id_and_votable_type_and_vote_scope" ON "votes" ("votable_id", "votable_type", "vote_scope")
```

As you can see Rails 4.2.9 and Rails 5 with migration tagged with [4.2], create same SQL, but Rails 5 with migration tagged with [5.0] creates timestamps with option `NOT NULL` (not bad actually) and a new index that may or may not be needed.

In order to have exactly the same database whether the user has Rails 4 or 5 we have to tag migrations with version [4.2].

A topic for another discussion is if we want to update migrations to do something different (add/remove indexes, etc)

**Comment by laertispappas** [comment]
@epergo Thank you for your feedback and your well explained details. So your suggestion is to lock the migration version on 4.2?

**Comment by epergo** [comment]
Yep, I think that would be the correct approach. What I am not sure about is how to test this.

To correctly test this functionality it would be necessary to create a dummy app, add the migration file and test if version was specified in class definition file but, is it worth it?

If you find any other way to do that and want to give a shot, go ahead. But I think this time we can skip tests.

**Comment by laertispappas** [comment]
@epergo Thank you once more for your feedback. I locked the migration to `4.2` when Rails 5 is detected to suppress deprecation warnings.  Unfortunately didn't find a quick way to test this...

**Comment by epergo** [comment]
Ok! I'm going to merge this. If you ever want to give it a try and test migrations, feel free to create a new pull request.

Thanks @laertispappas !

**Comment by ssaunier** [comment]
Awesome! Do you plan to push a new release to rubygems soon 📦 ?

**Comment by epergo** [comment]
Yes! That's the idea, I want to make a couple of changes more before releasing.

Meanwhile you can install it from branch `master` instead of current version on rubygems

File: lib/generators/acts_as_votable/migration/migration_generator.rb (hunk 1 above)
Comment (wilpar) [reply lib/generators/acts_as_votable/migration/migration_generator.rb]: rails 6 is live :)

**Pull request merged**